File: domain/src/main/java/no/nav/melosys/domain/behandlingsgrunnlag/soeknad/UtenlandskIdent.java
package no.nav.melosys.domain.behandlingsgrunnlag.soeknad;

import java.util.Objects;

public class UtenlandskIdent {
    public String ident;
    public String landkode;

    public UtenlandskIdent() {
    }

    public UtenlandskIdent(String ident, String landkode) {
        this.ident = ident;
        this.landkode = landkode;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        UtenlandskIdent that = (UtenlandskIdent) o;
        return ident.equals(that.ident) &&
            landkode.equals(that.landkode);
    }

    @Override
    public int hashCode() {
        return Objects.hash(ident, landkode);
    }
}

File: domain/src/main/kotlin/no/nav/melosys/domain/mottatteopplysninger/data/UtenlandskIdent.kt
package no.nav.melosys.domain.mottatteopplysninger.data

import java.util.*


class UtenlandskIdent {
    var ident: String? = null
    var landkode: String? = null

    constructor()
    constructor(ident: String?, landkode: String?) {
        this.ident = ident
        this.landkode = landkode
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as UtenlandskIdent
        return ident == that.ident && landkode == that.landkode
    }

    override fun hashCode(): Int {
        return Objects.hash(ident, landkode)
    }
}

